angelsrefining/prototypes/angels-functions.lua

```diff
@@ -400,10 +495,12 @@ function angelsmods.functions.create_gas_tech_icon(tints)
   -- allows a string of max 3 characters for default tints
   if tints then
     if type(tints) ~= "table" then
+      local reference = get_molecule_codes(tints)
+      ref = {reference[1].form,reference[2].form,reference[3].form} --don't go past the 3
       tints = {
-        top = unify_tint((icon_tint_table[string.sub(tints, 1, 1)] or {})[1]),
-        mid = unify_tint((icon_tint_table[string.sub(tints, 2, 2)] or {})[2]),
-        bot = unify_tint((icon_tint_table[string.sub(tints, 3, 3)] or {})[3])
+        top = unify_tint(icon_tints_table[ref[1]][1] or {}),
+        mid = unify_tint(icon_tints_table[ref[2]][2] or {}),
+        bot = unify_tint(icon_tints_table[ref[3]][3] or {})
       }

```

Comment:
The ref step is obsolete, it also does not provide a fallback if there are less than 3 refs defined?
```suggestion
      local reference = get_molecule_codes(tints)
      tints = {
        top = unify_tint(icon_tints_table[(reference[1] or {form="unknown"}).form][1] or {}),
        mid = unify_tint(icon_tints_table[(reference[2] or {form="unknown"}).form][2] or {}),
        bot = unify_tint(icon_tints_table[(reference[3] or {form="unknown"}).form][3] or {})
      }
```